app/src/main/java/app/wefridge/wefridge/datamodel/ItemController.kt

```diff
@@ -0,0 +1,117 @@
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.GeoPoint
+import java.lang.Exception
+import kotlin.collections.ArrayList
+
+class ItemController: ItemControllerInterface {
+    private val TAG = "ItemsOnFirebase"
+    private val db = FirebaseFirestore.getInstance()
+    private val items = ArrayList<Item>()
+
+    /*
+    * The function getItems is based on an example provided on
+    * https://firebase.google.com/docs/firestore/query-data/get-data
+    *
+    * The functions deleteItem, overrideItem and addItem are partially
+    * based on code snippets provided by the Firebase Documentation:
+    * https://firebase.google.com/docs/firestore/manage-data/add-data
+    * */
+    override fun getItems(callbackOnSuccess: (ArrayList<Item>) -> kotlin.Unit, callbackOnFailure: (Exception) -> kotlin.Unit) {
+        // TODO: only get items of specifc user and the groups participants
+        val ownerController: OwnerControllerInterface = OwnerController()
+        ownerController.getCurrentUser { owner ->
+            db.collection("items")
+                .whereEqualTo("owner", owner)
+                .get()
+                .addOnSuccessListener { itemDocuments ->
+                    for (item in itemDocuments) {
+                        try {
+                            items.add(parse(item.data, item.id))
+                        } catch (exc: ItemOwnerMissingException) {
+                            // TODO: consider to remove this, because requests already filters by owner
+                            item.data["owner"] = owner.id
+                            items.add(parse(item.data, item.id))
+                        }
+                    }
+
+                    callbackOnSuccess(items)
+                }
+
+                .addOnFailureListener { exception ->
+                    Log.w(TAG, "Error getting documents.", exception)
+                    callbackOnFailure(exception)
+                }
+        }
+    }
+
+    override fun deleteItem(item: Item) {
+        TODO("Not yet implemented")
+    }
+
+    override fun saveItem(item: Item, callbackOnSuccess: () -> kotlin.Unit, callbackOnFailure: (Exception) -> kotlin.Unit) {
+        // TODO: insert condition: when isShared == true location coordinates have to be != null!!!
+        if (item.firebaseId != null) overrideItem(item, { callbackOnSuccess() }, { exception -> callbackOnFailure(exception) })
+        else addItem(item, { callbackOnSuccess() }, { exception -> callbackOnFailure(exception) })
+    }
+
+    private fun overrideItem(item: Item, callbackOnSuccess: () -> kotlin.Unit, callbackOnFailure: (Exception) -> kotlin.Unit) {
+        if (item.firebaseId != null)
+            db.collection("items").document(item.firebaseId!!)
+                .set(item.getHashMap())
+                .addOnSuccessListener {
+                    Log.d(TAG, "item successfully written!")
```

Comment:
Maybe add item name in the log, also should not be item successfully written, but "Item successfully overridden"